backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -102,111 +108,103 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
             {
-                const MissionStatus NewStatus = MissionStatus.Failed;
-                _logger.LogWarning(
-                    "Mission run {MissionRunId} was not started successfully. Status updated to '{Status}'.\nReason: {FailReason}",
-                    missionRun.Id,
-                    NewStatus,
-                    ex.Message
-                );
-                missionRun.Status = NewStatus;
-                missionRun.StatusReason = $"Failed to start: '{ex.Message}'";
-                MissionService.Update(missionRun);
+                // We want to continue driving to a safe position if the isar state is idle
+                if (ex.IsarStatusCode != StatusCodes.Status409Conflict)
+                {
+                    _logger.LogError(ex, "Failed to stop the current mission on robot {RobotName} because: {ErrorMessage}", robot.Name, ex.Message);
+                    return;
+                }
+            }
+            catch (Exception ex)
+            {
+                string message = "Error in ISAR while stopping current mission, cannot drive to safe position";
+                _logger.LogError(ex, "{Message}", message);
+                return;
             }
-        }
 
-        private static bool MissionRunQueueIsEmpty(IList<MissionRun> missionRunQueue)
-        {
-            return !missionRunQueue.Any();
+            await MissionSchedulingService.ScheduleMissionToReturnToSafePosition(e.RobotId, e.AreaId);
         }
 
-        private async Task<bool> TheSystemIsAvailableToRunAMission(Robot robot, MissionRun missionRun)
+        private async void OnEmergencyButtonDepressedForRobot(object? sender, EmergencyButtonPressedForRobotEventArgs e)
         {
-            bool ongoingMission = await OngoingMission(robot.Id);
-
-            if (ongoingMission)
-            {
-                _logger.LogInformation("Mission run {MissionRunId} was not started as there is already an ongoing mission", missionRun.Id);
-                return false;
-            }
-            if (robot.Status is not RobotStatus.Available)
-            {
-                _logger.LogInformation("Mission run {MissionRunId} was not started as the robot is not available", missionRun.Id);
-                return false;
-            }
-            if (!robot.Enabled)
+            _logger.LogInformation("Triggered EmergencyButtonPressed event for robot ID: {RobotId}", e.RobotId);
+            var robot = await RobotService.ReadById(e.RobotId);
+            if (robot == null)
             {
-                _logger.LogWarning("Mission run {MissionRunId} was not started as the robot {RobotId} is not enabled", missionRun.Id, robot.Id);
-                return false;
+                _logger.LogError("Robot with ID: {RobotId} was not found in the database", e.RobotId);
+                return;
             }
-            if (missionRun.DesiredStartTime > DateTimeOffset.UtcNow)
+
+            var area = await AreaService.ReadById(e.AreaId);
```

Comment:
Should we not use the area information from the robot variable?

Reply:
Don't have a strong opinion on this, both methods should work